DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs

```diff
@@ -1,19 +1,39 @@
 ﻿namespace DigitalLearningSolutions.Web.ViewModels.MyAccount
 {
+    using DigitalLearningSolutions.Data.Models.CustomPrompts;
     using DigitalLearningSolutions.Data.Models.User;
 
     public class MyAccountViewModel
     {
         public MyAccountViewModel(
             AdminUser? adminUser,
-            DelegateUser? delegateUser)
+            DelegateUser? delegateUser,
+            CentreCustomPrompts? customPrompts)
         {
             FirstName = adminUser?.FirstName ?? delegateUser?.FirstName;
             Surname = adminUser?.LastName ?? delegateUser?.LastName;
             User = adminUser?.EmailAddress ?? delegateUser?.EmailAddress;
             ProfilePicture = adminUser?.ProfileImage ?? delegateUser?.ProfileImage;
             Centre = adminUser?.CentreName ?? delegateUser?.CentreName;
             DelegateNumber = delegateUser?.CandidateNumber;
+            JobGroup = delegateUser?.JobGroupName;
+
+            CustomField1 = PopulateCustomFieldViewModel(customPrompts?.CustomField1, delegateUser?.Answer1);
+            CustomField2 = PopulateCustomFieldViewModel(customPrompts?.CustomField2, delegateUser?.Answer2);
+            CustomField3 = PopulateCustomFieldViewModel(customPrompts?.CustomField3, delegateUser?.Answer3);
+            CustomField4 = PopulateCustomFieldViewModel(customPrompts?.CustomField4, delegateUser?.Answer4);
+            CustomField5 = PopulateCustomFieldViewModel(customPrompts?.CustomField5, delegateUser?.Answer5);
+            CustomField6 = PopulateCustomFieldViewModel(customPrompts?.CustomField6, delegateUser?.Answer6);
+        }
+
+        private CustomFieldViewModel? PopulateCustomFieldViewModel(CustomPrompt? customPrompt, string? answer)
+        {
+            if (customPrompt == null)
+            {
+                return null;
+            }
+
+            return new CustomFieldViewModel(customPrompt.CustomPromptText, customPrompt.Mandatory, answer);

```

Comment:
The ticket mentions that one of the things that should be displayed is the options - "Options is all potential options that should be displayed in a dropdown. If it’s null, we should display a text input." Are the options not being brought into the view model here for any particular reason?

Reply:
No dropdown in display mode as it is read only.
"The "options" on the ticket is a bit misleading because you don't need to display them at this stage"
They'll be used for the Edit Details page.